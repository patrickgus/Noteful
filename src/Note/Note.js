import React from "react";
import { Link } from "react-router-dom";
import { format, parseISO } from "date-fns";
import "./Note.css";
import AppContext from "../Context";

export default class Note extends React.Component {
  static contextType = AppContext;

  state = {
    error: null
  };

  handleDelete = async () => {
    try {
      if (this.props.match.params.noteId) {
        await this.context.onDeleteNote(this.props.id);
        this.props.history.push("/");
        this.context.updateNoteState(this.props.id);
      } else {
        await this.context.onDeleteNote(this.props.id);
        this.context.updateNoteState(this.props.id);
      }
    } catch (err) {
      this.setState({ error: err.message });
    }
  };

  render() {
    return (
      <div className="Note">
        <h2 className="Note__title">
          <Link to={`/note/${this.props.id}`}>{this.props.name}</Link>
        </h2>
        <button
          className="Note__delete"
          type="button"
          onClick={this.handleDelete}
        >
          remove
        </button>
        <div className="Note__dates">
          <div className="Note__dates-modified">
            Modified{" "}
            <span className="Date">
              {format(parseISO(this.props.modified), "do MMM yyyy")}
            </span>
          </div>
        </div>
      </div>
    );
  }
}
