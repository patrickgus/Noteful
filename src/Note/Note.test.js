import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Note from "./Note";

describe(`Note component`, () => {
  const newDate = new Date(2018, 12, 15);

  const props = {
    id: "a",
    name: "test-class-name",
    modified: newDate.toISOString()
  };

  it("renders the Note given props", () => {
    const wrapper = shallow(<Note {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
