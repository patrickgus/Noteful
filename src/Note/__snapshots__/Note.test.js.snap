// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders the Note given props 1`] = `
<div
  className="Note"
>
  <h2
    className="Note__title"
  >
    <Link
      to="/note/a"
    >
      test-class-name
    </Link>
  </h2>
  <button
    className="Note__delete"
    onClick={[Function]}
    type="button"
  >
    remove
  </button>
  <div
    className="Note__dates"
  >
    <div
      className="Note__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        15th Jan 2019
      </span>
    </div>
  </div>
</div>
`;
